test/common.js

```diff
@@ -12,6 +12,61 @@ exports.tmpDirName = 'tmp';
 exports.PORT = +process.env.NODE_COMMON_PORT || 12346;
 exports.isWindows = process.platform === 'win32';
 
+var rimrafSync = function(p) {
+  try {
+    var st = fs.lstatSync(p);
+  } catch (e) {
+    if (e.code === 'ENOENT')
+      return;
+  }
+
+  try {
+    if (st && st.isDirectory())
+      rmdirSync(p, null);
+    else
+      fs.unlinkSync(p);
+  } catch (e) {
+    if (e.code === 'ENOENT')
+      return;
+    if (e.code === 'EPERM')
+      return rmdirSync(p, er);
+    if (e.code !== 'EISDIR')
+      throw e;
+    rmdirSync(p, e);
+  }
+};
+
+var rmdirSync = function(p, originalEr) {
+  try {
+    fs.rmdirSync(p);
+  } catch (e) {
+    if (e.code === 'ENOENT')
+      return;
+    if (e.code === 'ENOTDIR')
+      throw originalEr;
+    if (e.code === 'ENOTEMPTY' || e.code === 'EEXIST' || e.code === 'EPERM') {
+      fs.readdirSync(p).forEach(function(f) {
+        rimrafSync(path.join(p, f));
+      });
+      fs.rmdirSync(p);
+    }
+  }
+};
+
+var refreshTmpDir = function() {
+  if (!process.send) { // Not a child process
+    try {
+      rimrafSync(exports.tmpDir);
+    } catch (e) {
+    }
+
+    try {
+      fs.mkdirSync(exports.tmpDir);
```

Comment:
(Replacing comment I just deleted because, uh, wrong topic. Let's try again.) Can we do that as a separate PR. As with the expose-refresh-and-put-in-individual-tests, I'm happy to do it and even excited to do it because it's The Right Thing to do. But I want to keep the scope of the individual change sets as narrow and manageable as possible. If that change *does* break something when jenkins is run, I wouldn't want that change to hold up the rest of the stuff that's here.